src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -173,37 +126,30 @@ private void CheckInvariants()
             Contract.ThrowIfFalse(_projectIdToProjectStateMap.Count == ProjectIds.Count);
             Contract.ThrowIfFalse(_projectIdToProjectStateMap.Count == _dependencyGraph.ProjectIds.Count);
 
-            // Only run this in debug builds; even the .Any() call across all projects can be expensive when there's a lot of them.
+            // Only run this in debug builds; even the .SetEquals() call across all projects can be expensive when there's a lot of them.
 #if DEBUG
-            // An id shouldn't point at a tracker for a different project.
-            Contract.ThrowIfTrue(_projectIdToTrackerMap.Any(kvp => kvp.Key != kvp.Value.ProjectState.Id));
-
             // project ids must be the same:
             Debug.Assert(_projectIdToProjectStateMap.Keys.SetEquals(ProjectIds));
             Debug.Assert(_projectIdToProjectStateMap.Keys.SetEquals(_dependencyGraph.ProjectIds));
 #endif
         }
 
-        private SolutionState Branch(
+        internal SolutionState Branch(
```

Comment:
became internal because there are algorithms in SOlutionCompilationState that want to do a bunch of operations, then branch the solution-state, then create the final SolutionCompilationState.

those agorithms are often updating the state of SolutionState and SolutionCompilationState in tandem.  So it's non-trivial to extract to an inner helper and an outer one.  It was just cleaner to expose Branch.